osbenchmark/workload_generator/workload_generator.py

```diff
@@ -42,10 +42,22 @@ def process_template(templates_path, template_filename, template_vars, output_pa
     with open(output_path, "w") as f:
         f.write(template.render(template_vars))
 
-
-def extract_mappings_and_corpora(client, output_path, indices_to_extract):
+def check_if_indices_and_number_of_docs_match(indices, number_of_docs, docs_were_requested):
+    '''
+    if key(s) and value(s) were provided to --number-of-docs,
+    the number of pairs need to be less than or equal to number of indices
+    '''
+    if docs_were_requested and len(indices) < len(number_of_docs):
+        raise exceptions.SystemSetupError("Number of key:value pairs exceeds number of indices mentioned in --indices. " +
+                                          "Ensure it is less than or equal to.")
```

Comment:
It would probably suffice to check if the index name in each K:V pair matches one of the indices in the `--indices` parameter and that would be more stringent.